tests/test_store.py

```diff
@@ -196,17 +227,62 @@ def test_view_query(
         columns: list,
         where_clauses: list,
         expected: str,
+        expected_new_query: bool,
     ) -> None:
-        # Ensure correct view is being utilized
+        # GIVEN a the correct fileview
         assert synapse_store_special_scope.storageFileview == "syn23643253"
 
+        # AND the approrpiate project scope
         synapse_store_special_scope.project_scope = project_scope
 
-        synapse_store_special_scope.query_fileview(columns, where_clauses)
-        # tests ._build_query()
-        assert synapse_store_special_scope.fileview_query == expected
-        # tests that the query was valid and successful, that a view subset has actually been retrived
-        assert synapse_store_special_scope.storageFileviewTable.empty is False
+        # WHEN the query is built and run
+        # THEN it should complete without raising an exception
+        with does_not_raise():
+            synapse_store_special_scope.query_fileview(columns, where_clauses)
+            # AND the query string should be as expected
+            assert synapse_store_special_scope.fileview_query == expected
+            # AND query should have recieved a non-empty table
+            assert synapse_store_special_scope.storageFileviewTable.empty is False
+            # AND the query should be new if expected
+            assert synapse_store_special_scope.new_query_different == expected_new_query
+
+    @pytest.mark.parametrize(
+        "asset_view,columns,message",
+        [
+            (
+                "syn62339865",
+                ["path"],
+                r"The path column has not been added to the fileview. .*",
+            ),
+            (
+                "syn62340177",
+                ["id"],
+                r"The columns id specified in the query do not exist in the fileview. .*",
+            ),
+        ],
+    )
+    def test_view_query_exception(
+        self,
+        synapse_store_special_scope: SynapseStorage,
+        asset_view: str,
+        columns: list[str],
+        message: str,
+    ) -> None:
+        project_scope = ["syn23643250"]
+
+        # GIVEN the appropriate test file view
+        CONFIG.synapse_master_fileview_id = asset_view
+        # AND the approrpiate project scope
+        synapse_store_special_scope.project_scope = project_scope
+
+        # WHEN the query is built and run
+        try:
+            # THEN it should raise a ValueError with the appropriate message
+            with pytest.raises(ValueError, match=message):
+                synapse_store_special_scope.query_fileview(columns)
+        finally:
+            # AND the fileview should be reset to the default so the other tests are not affected regardless of the outcome of the query
+            CONFIG.synapse_master_fileview_id = "syn23643253"
```

Comment:
If you are updating a global value and setting it back after you have the potential to cause side affects while the test is running before the global value was set back.

Rather than modifying the global value I would recommend to use some object matching here. That way any changes are only applicable within the `with` statement managed context.

I think you'll just need to patch the Configuration function for this property:
```
    @property
    def synapse_master_fileview_id(self) -> str:
        """
        Returns:
            str:
        """
        return self._synapse_config.master_fileview_id
```

Unfortunately mocking in Python can be a bit messy, but it's a trade-off to prevent side affects for other tests.

Reply:
@BryanFauble I'm encountering more issues with trying to mock the config object. Are this change and the others you noted with you approval necessary for the PR in the context of getting our fixes out quickly? I ask because I have other issues outstanding as well

Reply:
@thomasyu888 thanks for that! Yes I was concerned because as I was working on this PR to implement all the feedback it ended up taking much longer than I had estimated at our sprint planning and along with all the other coordination work affected my other issues. But I would appreciate the having the time to be able to actually implement things in a correct and robust way.

Reply:
As for challenges, I trust everyone puts in their effort, but don't be afraid to reach out for help you need it. What issues are you encountering?

Reply:
@thomasyu888 I was running into issues with how the mocked config was being used during the tests, but I think I was actually able to figure it out since I had the space to. The tests are passing locally now with the change so I think it's all good